dulwich/hooks.py

```diff
@@ -160,3 +160,28 @@ def clean_msg(success, *args):
 
         ShellHook.__init__(self, 'commit-msg', filepath, 1,
                            prepare_msg, clean_msg, controldir)
+
+class PostReceiveShellHook(ShellHook):
```

Comment:
You'll need an extra empty line above the class - this is probably why the tests are currently failing.